base/memory/shared_memory_posix.cc

```diff
@@ -90,11 +90,37 @@ bool SharedMemory::CreateAndMapAnonymous(size_t size) {
 // of mem_filename after FilePathForMemoryName().
 bool SharedMemory::Create(const SharedMemoryCreateOptions& options) {
   DCHECK(!shm_.IsValid());
-  if (options.size == 0) return false;
 
   if (options.size > static_cast<size_t>(std::numeric_limits<int>::max()))
     return false;
 
+#if defined(OS_ANDROID) && defined(CASTANETS)
+  // "name" is just a label in ashmem. It is visible in /proc/pid/maps.
+  int fd = ashmem_create_region(
+      options.name_deprecated ? options.name_deprecated->c_str() : "",
+      options.size);
+  shm_ = SharedMemoryHandle::ImportHandle(fd, options.size);
+  if (!shm_.IsValid()) {
+    DLOG(ERROR) << "Shared memory creation failed";
+    return false;
+  }
+
+  int flags = PROT_READ | PROT_WRITE | (options.executable ? PROT_EXEC : 0);
+  int err = ashmem_set_prot_region(shm_.GetHandle(), flags);
+  if (err < 0) {
+    DLOG(ERROR) << "Error " << err << " when setting protection of ashmem";
+    return false;
+  }
+
+  readonly_shm_ =
+      SharedMemoryHandle(FileDescriptor(dup(fd), false),
+                         options.size, shm_.GetGUID());
+  requested_size_ = options.size;
+
+  return true;
+#else
+  if (options.size == 0) return false;
```

Comment:
Why did you move this code from the original line?

Reply:
In shared_memory_android.cc, there is no validation for size == 0 case, so I moved this line below.
And it is also mentioned as below on line 296 on shared_memory_posix.cc, so I thought size = 0 while creating ashmem is valid. 

// On Android, Map can be called with a size and offset of zero to use the
// ashmem-determined size.